src/core/lib/security/security_connector/ssl/ssl_security_connector.cc

```diff
@@ -304,6 +305,7 @@ class grpc_ssl_server_security_connector
    * Current certificate config will continue to be used if the callback returns
    * an error. Returns true if new credentials were successfully loaded. */
   bool try_fetch_ssl_server_credentials() {
+    grpc_core::MutexLock lock(&mu_);
```

Comment:
How about moving it after L311?

Reply:
Good point. We can save some performance when `try_fetch_ssl_server_credentials` is called in `add_handshakers`.